File: rss-reader/app/src/main/java/com/abubusoft/kripton/examples/rssreader/persistence/RssDataSource.java
package com.abubusoft.kripton.examples.rssreader.persistence;

import com.abubusoft.kripton.android.annotation.BindDataSource;

@BindDataSource(fileName = "rss.db",daoSet = {DaoArticle.class, DaoChannel.class}, asyncTask = true)
public interface RssDataSource {
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/rssreader/service/persistence/RssDataSource.kt
package com.abubusoft.rssreader.service.persistence

import com.abubusoft.kripton.android.annotation.BindDataSource
import com.abubusoft.rssreader.service.persistence.BindRssDataSource.*

@BindDataSource(fileName = "rss.db", daoSet = [DaoRss::class, DaoArticle::class, DaoChannel::class], asyncTask = true)
interface RssDataSource


fun BindRssDataSource.execute(transition: Transaction) {
    return getInstance().execute(transition)
}